### DISCUSSION TITLE
Library size

### DISCUSSION THREAD

**Discussion by lllama** [pydantic/pydantic#3158] [Question] [answered]
Loving pydantic but I'm currently trying to include it in my lambda function in AWS and I'm hitting the 250Mb limit. It looks like 90Mb is taken up by pydantic. Is there any way to slim the package down?

Edit: it looks like the bulk of the size is taken up with the compiled .so files. Is there a way to install a non-compiled version? (I'll take the performance hit)

**Comment by samuelcolvin** [comment, marked as answer]
Only by rewriting it, or not compiling it.

One of those might happen in future, but not soon.

In the meantime, you need to install it without using the compiled binaries, it'll be around 50% slower but otherwise perform the same. Details on the install page of the docs.

**Reply by lllama** [reply]
Fair enough. I just installed using poetry and it looks like it's pulled in the compiled version from pypi. It looks like there's a wheel without the compiled libs, so I'll try and get that pulled in instead.

**Reply by samuelcolvin** [reply]
Exactly, see https://github.com/samuelcolvin/pydantic/issues/555#issuecomment-648005628

**Reply by lllama** [reply]
For anyone else who bumps up against this, the following in your requirements file will install the non-compiled version:
```
pydantic; sys_platform == "py3"
```
